tests/unit/conditions/test_rpc_condition.py

```diff
@@ -133,3 +136,98 @@ def test_rpc_condition_invalid_comparator_value_type(invalid_value, rpc_conditio
                 value=invalid_value,
             ),
         )
+
+
+def test_rpc_condition_uses_provided_endpoint(mocker):
+    # Mock HTTPProvider
+    mock_http_provider_spy = mocker.spy(HTTPProvider, "__init__")
+
+    # Mock eth module
+    mock_eth = mocker.Mock()
+    mock_eth.get_balance.return_value = 0
+    mock_eth.chain_id = 8453
+
+    # Create Web3 mock with required attributes
+    mock_w3 = mocker.Mock()
+    mock_w3.eth = mock_eth
+    mock_w3.middleware_onion = mocker.Mock()
```

Comment:
You can collapse this into:
```suggestion
    # Mock eth module
    mock_w3 = mocker.Mock()
    mock_w3.eth.get_balance.return_value = 0
    mock_w3.eth.chain_id = 8453
```